src/controller/python/chip-device-ctrl.py

```diff
@@ -719,8 +734,228 @@ def do_EOF(self, line):
     def emptyline(self):
         pass
 
+###  Additions needed by the Test Harness Tool ###
+# TODO: Implement a custom device manager instead of using the existing manager object
+# https://github.com/chip-csg/connectedhomeip/issues/8
+device_manager = DeviceMgrCmd(rendezvousAddr=None,
+                             controllerNodeId=0, bluetoothAdapter=0)
+
+
+# CHIP commands needed by the Harness Tool
+def echo_alive(message):
+    print(message)
+    return message
+
+def resolve(fabric_id: str, node_id: int) -> Dict[str, Any]:
+    try:
+        __check_supported_os()
+        err = device_manager.devCtrl.ResolveNode(int(fabric_id), node_id)
+        if err != 0:
+            return __get_response_dict(status = StatusCodeEnum.FAILED, error = f"Failed to resolve node, with error code: {err}")
+
+        address = device_manager.devCtrl.GetAddressAndPort(node_id)
+        if address is not None:
+            address = "{}:{}".format(
+                *address)  
+            return __get_response_dict(status = StatusCodeEnum.SUCCESS, result = {'address': address}) 
+            
+    except Exception as e:
+        return __get_response_dict(status = StatusCodeEnum.FAILED, error = str(e))
+
+def zcl_add_network(node_id: int, ssid: str, password: str, endpoint_id: Optional[int] = 1, group_id: Optional[int] = 0, breadcrumb: Optional[int] = 0, timeoutMs: Optional[int] = 1000) -> Dict[str, Any] :
```

Comment:
Can you make this function work for both adding thread and wifi networks, rather than two different functions.

Reply:
Done.